packages/core/snapshot/snapshot.go

```diff
@@ -1,49 +1,47 @@
 package snapshot
 
 import (
-	"bytes"
-	"context"
 	"fmt"
 	"os"
 
-	"github.com/cockroachdb/errors"
-	"github.com/iotaledger/hive.go/marshalutil"
-	"github.com/iotaledger/hive.go/serix"
-	"github.com/iotaledger/hive.go/stringify"
-
 	"github.com/iotaledger/goshimmer/packages/core/epoch"
 	"github.com/iotaledger/goshimmer/packages/core/ledger"
 	"github.com/iotaledger/goshimmer/packages/core/ledger/vm/devnetvm"
 	"github.com/iotaledger/goshimmer/packages/core/mana"
-	"github.com/iotaledger/goshimmer/packages/core/notarization"
-	"github.com/iotaledger/goshimmer/packages/core/tangleold"
+	"github.com/iotaledger/hive.go/serix"
 )
 
 // Snapshot contains the data to be put in a snapshot file.
 type Snapshot struct {
 	LedgerSnapshot *ledger.Snapshot
 }
 
-// CreateStreamableSnapshot creates a snapshot file to the given file path.
-func CreateStreamableSnapshot(filePath string, t *tangleold.Tangle, nmgr *notarization.Manager) (*ledger.SnapshotHeader, error) {
-	f, err := os.Create(filePath)
+func init() {
+	typeSet := new(serix.TypeSettings)
+	ts := typeSet.WithLengthPrefixType(serix.LengthPrefixTypeAsUint32)
+
+	err := serix.DefaultAPI.RegisterTypeSettings([]*ledger.OutputWithMetadata{}, ts)
 	if err != nil {
-		return nil, fmt.Errorf("fail to create snapshot file: %s", err)
+		panic(fmt.Errorf("error registering OutputWithMetadata slice type settings: %w", err))
 	}
 
-	outputWithMetadataProd := NewLedgerOutputWithMetadataProducer(t.Ledger)
-	committableEC, fullEpochIndex, err := t.Options.CommitmentFunc()
+	err = serix.DefaultAPI.RegisterTypeSettings(map[epoch.Index]*ledger.EpochDiff{}, ts)
 	if err != nil {
-		return nil, err
+		panic(fmt.Errorf("error registering EpochDiff map type settings: %w", err))
 	}
+}
 
-	header := &ledger.SnapshotHeader{
-		FullEpochIndex: fullEpochIndex,
-		DiffEpochIndex: committableEC.EI(),
-		LatestECRecord: committableEC,
+// CreateSnapshot creates a snapshot file to the given file path.
+func CreateSnapshot(filePath string,
```

Comment:
Through the producers you are extracting data over a certain amount of time; as you are not locking the notarization manager in between this producer calls, it could be possible that your header will contain data that won't match your UTXOs or Diffs. Imagine for instance that a new epoch is confirmed in between asking for the latest confirmed epoch and your epoch diffs: you will have unmatched header and snapshot contents!
I think you should make sure that the ledgerstate and the diffs of the notarization manager do not change after you determine what epochs you are going to dump into the snapshot: 1) by locking the ledgerstate so it cannot be updated in the meantime, 2) passing the epochs you want the diffs for to the producers calls.